src/Microsoft.AspNet.Razor.Runtime/TagHelpers/ReadOnlyTagHelperAttributeList.cs

```diff
@@ -78,18 +70,20 @@ public ReadOnlyTagHelperAttributeList(IEnumerable<TAttribute> attributes)
                     throw new ArgumentNullException(nameof(name));
                 }
 
-                return Attributes.FirstOrDefault(attribute => NameEquals(name, attribute));
+                for (var i = 0; i < Attributes.Count; i++)
```

Comment:
I generally add a `// Perf: Avoid allocating enumerator` in these kind of micro things